### PR TITLE
[server] fix goroutine leak

### LINKED ISSUE #179: Memory usage 84G
I'm run dogechain node, Synchronized to the latest block.
The memory utilization rate has gradually increased to 84G.
This is far from the official recommendation. How can I limit the memory usage?

vCPUs >= 4
Memory: at least 8GB
Disk space
at least 200GB for the data directory ( recommend SSD )
data disk mount to /data

CONTAINER ID   NAME          CPU %     MEM USAGE / LIMIT     MEM %     NET I/O           BLOCK I/O         PIDS
d20ef9699492   dogechain         0.72%     83.99GiB / 249.2GiB   33.71%    17.6GB / 19.58GB   1.85GB / 10.16TB   39

### PR DESCRIPTION
**Pull request by 0xcb9ff9**
# Description

- `blockchain.Subscription` interface method  `GetEventCh` each use will create a goroutine, easy to misuse, like [protocol/syncer.go#L157](https://github.com/0xcb9ff9/dogechain/blob/a530848fa968851069ccaca533a912208161fdea/protocol/syncer.go#L157), this PR remove this method use `GetEvent` replace

- there is a risk of blocking [helper/progress/chain.go#L112
](https://github.com/dogechain-lab/dogechain/blob/a530848fa968851069ccaca533a912208161fdea/helper/progress/chain.go#L112)
- grpc client goroutine leak
-  use [BasicConnMgr](https://pkg.go.dev/github.com/libp2p/go-libp2p@v0.18.0/p2p/net/connmgr#NewConnManager) replace temporary dial, bootnode and static node address is [permanent](https://pkg.go.dev/github.com/libp2p/go-libp2p-core@v0.14.0/peerstore#PermanentAddrTTL) valid

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [x] Hotfix (change that solves an urgent issue, and requires immediate attention)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels

## Testing

- [x] I have tested this code with the official test suite

### CODE REVIEW

File: protocol/client.go [hunk 1]
```diff
@@ -243,31 +263,32 @@ func (client *syncPeerClient) startNewBlockProcess() {
...
+			case client.peerConnectionUpdateCh <- e:
 			}
-
-			client.peerConnectionUpdateCh <- e
 		}
 	}
 }
 
-// CloseStream closes stream
-func (client *syncPeerClient) CloseStream(peerID peer.ID) error {
```
Comment (0xcb9ff9) [inline protocol/client.go:267]: `CloseStream` only closing when the connection is disconnected, and should not be handed over to the user. This design can reduce the complexity of connection management

File: protocol/client.go [hunk 2]
```diff
@@ -370,21 +391,20 @@ func (client *syncPeerClient) newSyncPeerClient(peerID peer.ID) (proto.V1Client,
...
 
-			return nil, fmt.Errorf("failed to open a stream, err %w", err)
-		}
+	// create new connection
+	conn, err := client.network.NewProtoConnection(_syncerV1, peerID)
+	if err != nil {
+		client.network.ForgetPeer(peerID, "not support syncer v1 protocol")
 
-		// save protocol stream
-		client.network.SaveProtocolStream(_syncerV1, conn, peerID)
```
Comment (0xcb9ff9) [inline protocol/client.go:386]: remove `SaveProtocolStream`, create and save from `NewProtoConnection`, only single object, hide detail

**Review by DarianShawn — APPROVED** [review]
Thanks for the great work.
LGTM

**Review by abrahamcruise321 — APPROVED** [review]
LGTM. Thx!

**Pull request merged**